src/utils/networkHelper.js

```diff
@@ -69,72 +71,74 @@ const viewTicketsAccordingToStatus = async (status, toggleErrorState) => {
-const queryWithEquality = async (collection, field, value, toggleErrorState) => {
+const queryWithEquality = async (collectionName, field, value, toggleErrorState) => {
   try {
-    const snapshot = await db.collection(collection).where(field, '==', value).get();
-    return snapshot;
+    const collectionRef = collection(db, collectionName);
+    const q = query(collectionRef, where(field, '==', value));
+    const snapshot = await getDocs(q);
+    return snapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
   } catch (error) {
-    toggleErrorState(`Error querying ${collection} by ${field} == ${value}: ${error.message}`,true);
-    throw new Error(`Error querying ${collection} by ${field} == ${value}: ${error.message}`);
+    toggleErrorState(`Error querying ${collectionName} by ${field} == ${value}: ${error.message}`, true);
+    throw new Error(`Error querying ${collectionName} by ${field} == ${value}: ${error.message}`);
   }
 };
 
 // Fetch a document by ID
-const fetchDocument = async (collection, docId, toggleErrorState) => {
+const fetchDocument = async (collectionName, docId, toggleErrorState) => {
   try {
-    const doc = await db.collection(collection).doc(docId).get();
-    if (!doc.exists) {
-      throw new Error(`No document found with ID ${docId} in collection ${collection}`);
+    const docRef = doc(db, collectionName, docId);
+    const docSnap = await getDoc(docRef);
+    if (!docSnap.exists()) {
+      throw new Error(`No document found with ID ${docId} in collection ${collectionName}`);
     }
-    return doc;
+    return { id: docSnap.id, ...docSnap.data() };
   } catch (error) {
-    toggleErrorState(`Error fetching document ${docId} from ${collection}: ${error.message}`,true);
-    throw new Error(`Error fetching document ${docId} from ${collection}: ${error.message}`);
+    toggleErrorState(`Error fetching document ${docId} from ${collectionName}: ${error.message}`, true);
+    throw new Error(`Error fetching document ${docId} from ${collectionName}: ${error.message}`);
   }
 };
 
 // Fetch all documents in a collection
-const fetchAllDocumentsInCollection = async (collection, toggleErrorState) => {
+const fetchAllDocumentsInCollection = async (collectionName, toggleErrorState) => {
   try {
-    const snapshot = await db.collection(collection).get();
-    return snapshot;
+    const collectionRef = collection(db, collectionName);
+    const snapshot = await getDocs(collectionRef);
+    return snapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
   } catch (error) {
-    toggleErrorState(`Error fetching documents from collection ${collection}: ${error.message}`,true);
-    throw new Error(`Error fetching documents from collection ${collection}: ${error.message}`);
+    toggleErrorState(`Error fetching documents from collection ${collectionName}: ${error.message}`, true);
+    throw new Error(`Error fetching documents from collection ${collectionName}: ${error.message}`);
```

Comment:
Why are we throwing Error here ? what is the purpose of catch if we are throwing error even if we know we need to handle the error on ui here?

Reply:
Done